benches/distributions.rs

```diff
@@ -108,30 +108,28 @@ distr_int!(distr_poisson, u64, Poisson::new(4.0));
 
 // construct and sample from a range
 macro_rules! gen_range_int {
-    ($fnn:ident, $ty:ty, $low:expr, $high:expr) => {
+    ($fnn:ident, $ty:ty, $low:expr, $high:expr, $incr:expr) => {
         #[bench]
         fn $fnn(b: &mut Bencher) {
             let mut rng = XorShiftRng::new();
-            let mut high = $high;
-            let fake_increment = 0;
 
             b.iter(|| {
+                let mut high = $high;
                 let mut accum: $ty = 0;
                 for _ in 0..::RAND_BENCH_N {
                     accum = accum.wrapping_add(rng.gen_range($low, high));
-                    high += fake_increment;
+                    high += $incr;  // force recalculation of range each time
```

Comment:
How about `high = high.wrapping_add(1) & std::%ty::MAX`, so it also works for `i8`? (Not thought deeply about it to be honest)

Reply:
I think that would still wrap, and end up less than `low`? Current version seems to work anyway.